numpy/_core/src/umath/_scaled_float_dtype.c

```diff
@@ -787,8 +787,10 @@ sfloat_stable_sort_loop(
 {
     assert(data[0] == data[1]);
     assert(strides[0] == sizeof(npy_float64) && strides[1] == sizeof(npy_float64));
+#ifndef NDEBUG
     PyArrayMethod_SortParameters *parameters = (PyArrayMethod_SortParameters *)context->parameters;
     assert(parameters->flags == NPY_SORT_STABLE);
```

Comment:
Absolute nit, but to avoid distraction (fewer irrelevant lines), could we not just do
```
assert(((PyArrayMethod_SortParameters *)context->parameters)->flags == NPY_SORT_STABLE);
```

Reply:
Could be worth it given there's 2 more lines now!